### PR TITLE
Skip "Press X" screen when navigating back to main menu from another scene

### PR DESCRIPTION
**Pull request by trudeaua**
PlayerUIControlSpawn changed to wait for the scene to be done loading and stop the "should not get here" spam message. The message is still logged a few times but it no longer spams endlessly.

### CODE REVIEW

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs [hunk 1]
```diff
@@ -1,15 +1,35 @@
-﻿using UnityEngine;
+﻿using System.Collections;
+using UnityEngine;
 
 public class PlayerUIControlSpawn : MonoBehaviour
 {
+    [Tooltip("Seconds to wait for game state to not be loading")]
+    public float timeout = 10;
```
Comment (camvr) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs]: Why do we need a timeout for loading? Were there problems with loading lasting too long?

Comment (camvr) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs]: If you're just waiting for loading to be done, then we could potentially use events instead (which would certainly be cleaner)

Comment (trudeaua) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs]: Ok I'll switch to using events!

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs [hunk 2]
```diff
@@ -7,29 +7,16 @@ public class PlayerUIControlSpawn : MonoBehaviour
     public float timeout = 10;
     private void Start()
     {
-        StartCoroutine(Spawn());
+        LevelManager.Instance.loadingDoneEvent.AddListener(Spawn);
     }
-    private IEnumerator Spawn()
+    private void Spawn()
     {
```
Comment (camvr) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs]: Slightly nit-picky of me, but I've noticed in most files that you change the newlines between function definitions disappear... Not sure if its intentional or some weird automatic git line ending conversion?

File: Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs (hunk 1 above)
Comment (camvr) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/Game/PlayerUIControlSpawn.cs]: Thanks for changing it to use an event listener... Much cleaner. Don't think there's any reason for the `timeout` variable to hang around now

**Comment by trudeaua** [comment]
Thought I removed that variable... whoops

And I'll look into that newlines thing, dunno why its happening

**Pull request merged**